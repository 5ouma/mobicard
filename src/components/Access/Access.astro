---
import { Icon } from "astro-icon/components";

import "../../styles/global.css";

export type Props = { homepage: string; email?: string };
const { homepage, email } = Astro.props;

const url = new URL(homepage);
---

<div>
  <a
    href={homepage}
    target="_blank"
    rel="noopener noreferrer"
    aria-label={`Visit ${url.hostname}`}
  >
    <Icon name="mingcute:home-3-line" />
    <span aria-hidden="true">
      {`${url.hostname}${url.pathname === "/" ? "" : url.pathname}`}
    </span>
  </a>
  {email &&
  <a href=`mailto:${email}` aria-label={`Send an email to ${email}`}>
    <Icon name="mingcute:mail-line" />
    <span aria-hidden="true">{email}</span>
  </a>
  }
</div>

<style>
  div {
    display: flex;
    flex-direction: column;
    align-items: flex-start;
    gap: 0.3rem;
  }

  a {
    display: flex;
    justify-content: center;
    align-items: center;
    gap: 0.5rem;
    text-decoration: none;
    color: var(--foreground);
    border-bottom: 2px solid
      color-mix(in srgb, var(--foreground) 30%, transparent);
    transition: border-color 0.3s ease-in-out;

    &:hover,
    &:focus-visible {
      border-color: color-mix(in srgb, var(--foreground) 60%, transparent);
    }
  }

  svg {
    width: 1.5rem;
    height: auto;
    opacity: 80%;
  }

  span {
    font-family: "LINE Seed JP", sans-serif;
    font-size: 1rem;
    font-style: normal;
    font-weight: normal;
    line-height: normal;
  }
</style>
