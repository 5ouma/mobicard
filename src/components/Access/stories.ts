import type { Props } from "./Access.astro";

export const Default: Props = {
  homepage: new URL("https://example.com"),
  email: "email@example.com",
};

export const SubDomain: Props = {
  homepage: new URL("https://www.example.com"),
  email: "email@www.example.com",
};

export const WithoutEmail: Props = {
  homepage: new URL("https://www.example.com"),
};
