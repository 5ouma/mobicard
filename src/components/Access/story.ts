import type { Props } from "./Access.astro";

export const Default: Props = {
  homepage: "https://example.com",
  email: "email@example.com",
};

export const SubDomain: Props = {
  homepage: "https://www.example.com",
  email: "email@www.example.com",
};

export const WithoutEmail: Props = {
  homepage: "https://www.example.com",
};
