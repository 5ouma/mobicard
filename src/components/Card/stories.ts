import type { Props } from "./Card.astro";

export const Default: Props = { url: new URL("https://example.com") };
export const SubDomain: Props = { url: new URL("https://www.example.com") };
