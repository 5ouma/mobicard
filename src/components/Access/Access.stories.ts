import Access from "./Access.astro";
import * as stories from "./stories.ts";

export default {
  title: "Homepage and Email",
  component: Access,
};

export const Default = { args: stories.Default };
export const SubDomain = { args: stories.SubDomain };
export const WithoutEmail = { args: stories.WithoutEmail };
