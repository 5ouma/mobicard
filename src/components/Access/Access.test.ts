import { experimental_AstroContainer as AstroContainer } from "astro/container";
import { describe, expect, test } from "vitest";

import Access from "./Access.astro";
import * as stories from "./stories.ts";

describe("Access", () => {
  for (const [name, props] of Object.entries(stories)) {
    test(name, async () => {
      const container: AstroContainer = await AstroContainer.create();
      const result: string = await container.renderToString(Access, {
        props,
      });

      expect(result).toMatchSnapshot();
    });
  }
});
