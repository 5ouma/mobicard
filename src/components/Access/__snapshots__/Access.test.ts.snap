// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Access > Default 1`] = `"<div data-astro-cid-j4povljh> <a href="https://example.com/" target="_blank" rel="noopener noreferrer" aria-label="Visit example.com" data-astro-cid-j4povljh> <svg width="1em" height="1em" data-astro-cid-j4povljh="true" data-icon="mingcute:home-3-line">   <symbol id="ai:mingcute:home-3-line" viewBox="0 0 24 24"><g fill="none"><path d="m12.593 23.258l-.011.002l-.071.035l-.02.004l-.014-.004l-.071-.035q-.016-.005-.024.005l-.004.01l-.017.428l.005.02l.01.013l.104.074l.015.004l.012-.004l.104-.074l.012-.016l.004-.017l-.017-.427q-.004-.016-.017-.018m.265-.113l-.013.002l-.185.093l-.01.01l-.003.011l.018.43l.005.012l.008.007l.201.093q.019.005.029-.008l.004-.014l-.034-.614q-.005-.018-.02-.022m-.715.002a.02.02 0 0 0-.027.006l-.006.014l-.034.614q.001.018.017.024l.015-.002l.201-.093l.01-.008l.004-.011l.017-.43l-.003-.012l-.01-.01z"/><path fill="currentColor" d="M10.772 2.688a2 2 0 0 1 2.317-.099l.139.1l8.384 6.52c.721.561.37 1.692-.499 1.785l-.116.006H20v8a2 2 0 0 1-1.85 1.995L18 21H6a2 2 0 0 1-1.994-1.85L4 19v-8h-.997C2.09 11 1.671 9.892 2.3 9.285l.088-.076zM12 4.267L5.625 9.225c.229.185.375.468.375.785V19h3v-5a3 3 0 1 1 6 0v5h3v-8.99c0-.317.146-.6.375-.785zM12 13a1 1 0 0 0-1 1v5h2v-5a1 1 0 0 0-1-1"/></g></symbol><use href="#ai:mingcute:home-3-line"></use>  </svg> <span aria-hidden="true" data-astro-cid-j4povljh> example.com </span> </a> <a href="mailto:email@example.com" aria-label="Send an email to email@example.com" data-astro-cid-j4povljh> <svg width="1em" height="1em" data-astro-cid-j4povljh="true" data-icon="mingcute:mail-line">   <symbol id="ai:mingcute:mail-line" viewBox="0 0 24 24"><g fill="none"><path d="m12.593 23.258l-.011.002l-.071.035l-.02.004l-.014-.004l-.071-.035q-.016-.005-.024.005l-.004.01l-.017.428l.005.02l.01.013l.104.074l.015.004l.012-.004l.104-.074l.012-.016l.004-.017l-.017-.427q-.004-.016-.017-.018m.265-.113l-.013.002l-.185.093l-.01.01l-.003.011l.018.43l.005.012l.008.007l.201.093q.019.005.029-.008l.004-.014l-.034-.614q-.005-.018-.02-.022m-.715.002a.02.02 0 0 0-.027.006l-.006.014l-.034.614q.001.018.017.024l.015-.002l.201-.093l.01-.008l.004-.011l.017-.43l-.003-.012l-.01-.01z"/><path fill="currentColor" d="M20 4a2 2 0 0 1 1.995 1.85L22 6v12a2 2 0 0 1-1.85 1.995L20 20H4a2 2 0 0 1-1.995-1.85L2 18V6a2 2 0 0 1 1.85-1.995L4 4zm0 3.414l-6.94 6.94a1.5 1.5 0 0 1-2.12 0L4 7.414V18h16zM18.586 6H5.414L12 12.586z"/></g></symbol><use href="#ai:mingcute:mail-line"></use>  </svg> <span aria-hidden="true" data-astro-cid-j4povljh>email@example.com</span> </a> </div> "`;

exports[`Access > SubDomain 1`] = `"<div data-astro-cid-j4povljh> <a href="https://www.example.com/" target="_blank" rel="noopener noreferrer" aria-label="Visit www.example.com" data-astro-cid-j4povljh> <svg width="1em" height="1em" data-astro-cid-j4povljh="true" data-icon="mingcute:home-3-line">   <symbol id="ai:mingcute:home-3-line" viewBox="0 0 24 24"><g fill="none"><path d="m12.593 23.258l-.011.002l-.071.035l-.02.004l-.014-.004l-.071-.035q-.016-.005-.024.005l-.004.01l-.017.428l.005.02l.01.013l.104.074l.015.004l.012-.004l.104-.074l.012-.016l.004-.017l-.017-.427q-.004-.016-.017-.018m.265-.113l-.013.002l-.185.093l-.01.01l-.003.011l.018.43l.005.012l.008.007l.201.093q.019.005.029-.008l.004-.014l-.034-.614q-.005-.018-.02-.022m-.715.002a.02.02 0 0 0-.027.006l-.006.014l-.034.614q.001.018.017.024l.015-.002l.201-.093l.01-.008l.004-.011l.017-.43l-.003-.012l-.01-.01z"/><path fill="currentColor" d="M10.772 2.688a2 2 0 0 1 2.317-.099l.139.1l8.384 6.52c.721.561.37 1.692-.499 1.785l-.116.006H20v8a2 2 0 0 1-1.85 1.995L18 21H6a2 2 0 0 1-1.994-1.85L4 19v-8h-.997C2.09 11 1.671 9.892 2.3 9.285l.088-.076zM12 4.267L5.625 9.225c.229.185.375.468.375.785V19h3v-5a3 3 0 1 1 6 0v5h3v-8.99c0-.317.146-.6.375-.785zM12 13a1 1 0 0 0-1 1v5h2v-5a1 1 0 0 0-1-1"/></g></symbol><use href="#ai:mingcute:home-3-line"></use>  </svg> <span aria-hidden="true" data-astro-cid-j4povljh> www.example.com </span> </a> <a href="mailto:email@www.example.com" aria-label="Send an email to email@www.example.com" data-astro-cid-j4povljh> <svg width="1em" height="1em" data-astro-cid-j4povljh="true" data-icon="mingcute:mail-line">   <symbol id="ai:mingcute:mail-line" viewBox="0 0 24 24"><g fill="none"><path d="m12.593 23.258l-.011.002l-.071.035l-.02.004l-.014-.004l-.071-.035q-.016-.005-.024.005l-.004.01l-.017.428l.005.02l.01.013l.104.074l.015.004l.012-.004l.104-.074l.012-.016l.004-.017l-.017-.427q-.004-.016-.017-.018m.265-.113l-.013.002l-.185.093l-.01.01l-.003.011l.018.43l.005.012l.008.007l.201.093q.019.005.029-.008l.004-.014l-.034-.614q-.005-.018-.02-.022m-.715.002a.02.02 0 0 0-.027.006l-.006.014l-.034.614q.001.018.017.024l.015-.002l.201-.093l.01-.008l.004-.011l.017-.43l-.003-.012l-.01-.01z"/><path fill="currentColor" d="M20 4a2 2 0 0 1 1.995 1.85L22 6v12a2 2 0 0 1-1.85 1.995L20 20H4a2 2 0 0 1-1.995-1.85L2 18V6a2 2 0 0 1 1.85-1.995L4 4zm0 3.414l-6.94 6.94a1.5 1.5 0 0 1-2.12 0L4 7.414V18h16zM18.586 6H5.414L12 12.586z"/></g></symbol><use href="#ai:mingcute:mail-line"></use>  </svg> <span aria-hidden="true" data-astro-cid-j4povljh>email@www.example.com</span> </a> </div> "`;

exports[`Access > WithoutEmail 1`] = `"<div data-astro-cid-j4povljh> <a href="https://www.example.com/" target="_blank" rel="noopener noreferrer" aria-label="Visit www.example.com" data-astro-cid-j4povljh> <svg width="1em" height="1em" data-astro-cid-j4povljh="true" data-icon="mingcute:home-3-line">   <symbol id="ai:mingcute:home-3-line" viewBox="0 0 24 24"><g fill="none"><path d="m12.593 23.258l-.011.002l-.071.035l-.02.004l-.014-.004l-.071-.035q-.016-.005-.024.005l-.004.01l-.017.428l.005.02l.01.013l.104.074l.015.004l.012-.004l.104-.074l.012-.016l.004-.017l-.017-.427q-.004-.016-.017-.018m.265-.113l-.013.002l-.185.093l-.01.01l-.003.011l.018.43l.005.012l.008.007l.201.093q.019.005.029-.008l.004-.014l-.034-.614q-.005-.018-.02-.022m-.715.002a.02.02 0 0 0-.027.006l-.006.014l-.034.614q.001.018.017.024l.015-.002l.201-.093l.01-.008l.004-.011l.017-.43l-.003-.012l-.01-.01z"/><path fill="currentColor" d="M10.772 2.688a2 2 0 0 1 2.317-.099l.139.1l8.384 6.52c.721.561.37 1.692-.499 1.785l-.116.006H20v8a2 2 0 0 1-1.85 1.995L18 21H6a2 2 0 0 1-1.994-1.85L4 19v-8h-.997C2.09 11 1.671 9.892 2.3 9.285l.088-.076zM12 4.267L5.625 9.225c.229.185.375.468.375.785V19h3v-5a3 3 0 1 1 6 0v5h3v-8.99c0-.317.146-.6.375-.785zM12 13a1 1 0 0 0-1 1v5h2v-5a1 1 0 0 0-1-1"/></g></symbol><use href="#ai:mingcute:home-3-line"></use>  </svg> <span aria-hidden="true" data-astro-cid-j4povljh> www.example.com </span> </a>  </div> "`;
